### DISCUSSION TITLE
kiali pods not coming up with custom url context.

### DISCUSSION THREAD

**Discussion by rohit131989** [kiali/kiali#5956] [Q&A]
I am trying to update context of kiali from '/kiali' to 'kialidashboard'.
For this, I did following changes in the config map of kiali

    server:
      address: ''
      port: 20001
      web_fqdn: 'kiali.istio-system.svc.cluster.local'
      web_history_mode: ''
      web_port: '20001'
      web_root: /kialidashboard
      web_schema: 'http'

But after restarting the pod, the kiali pod is not coming up.

Tail logs are as follows:

W0329 11:43:25.079481    5236 gcp.go:119] WARNING: the gcp auth plugin is deprecated in v1.22+, unavailable in v1.26+; use gcloud instead.
To learn more, consult https://cloud.google.com/blog/products/containers-kubernetes/kubectl-auth-changes-in-gke
{"level":"info","time":"2023-03-29T11:42:18Z","message":"Adding a RegistryRefreshHandler"}
2023-03-29T11:42:18Z INF Kiali: Version: v1.65.0, Commit: 312d1b55709d39b8afcc40d07e392de71bf3433e, Go: 1.19.6
2023/03/29 11:42:18 maxprocs: Leaving GOMAXPROCS=16: CPU quota undefined
2023-03-29T11:42:18Z INF Using authentication strategy [token]
2023-03-29T11:42:18Z INF Some validation errors will be ignored [KIA1201]. If these errors do occur, they will still be logged. If you think the validation errors you see are incorrect, please report them to the Kiali team if you have not done so already and provide the details of your scenario. This will keep Kiali validations strong for the whole community.
2023-03-29T11:42:18Z INF Initializing Kiali Cache
2023-03-29T11:42:18Z INF Adding a RegistryRefreshHandler
2023-03-29T11:42:18Z INF [Kiali Cache] Waiting for cluster-scoped cache to sync
2023-03-29T11:42:18Z INF [Kiali Cache] Started
2023-03-29T11:42:18Z INF [Kiali Cache] Kube cache is active for namespaces [.*]
2023-03-29T11:42:18Z INF Server endpoint will start at [:20001/kialidashboard]
2023-03-29T11:42:18Z INF Server endpoint will serve static content from [/opt/kiali/console]
2023-03-29T11:42:18Z INF Starting Metrics Server on [:9090]

Could it be an issue that the Server endpoint part in logs is showing [/opt/kiali/console] ? If so, how do I fix it

**Comment by jmazzitelli** [comment]
_moving this to Q&A discussions_

do a `kubectl describe` on the pod and see what information it tells you. Sounds like the liveness probe is not passing.

**Reply by rohit131989** [reply]
Thanks @jmazzitelli 
The liveness probe was getting failed due to health url having /kiali as path.

I btw installed kiali using helm with kiali operator as below:

helm install \
    --set cr.create=true \
    --set cr.namespace=istio-system \
    --namespace kiali-operator \
    --create-namespace \
    kiali-operator \
    kiali/kiali-operator

Could you please suggest how to pass kiali config related values in this command.

**Reply by jmazzitelli** [reply]
> Could you please suggest how to pass kiali config related values in this command.

The kiali.io documentation will help. It should tell you what you need to know:

- installation docs: https://kiali.io/docs/installation/installation-guide/install-with-helm/#install-with-operator
- Kiali CR reference with all the settings: https://kiali.io/docs/configuration/kialis.kiali.io/ (the setting you are interested is [this one](https://kiali.io/docs/configuration/kialis.kiali.io/#.spec.server.web_root)) 
- updating the Kiali CR (useful to update an existing Kiali with a new configuration without re-installing the operator): https://kiali.io/docs/installation/installation-guide/creating-updating-kiali-cr/

In short, you want to pass `--set cr.spec.server.web_root=/kialidashboard` to your helm command.

When you change that web_root setting in the Kiali CR (or pass it to the operator helm chart via the `cr.spec.server.web_root` setting), the operator makes sure the liveness probe and readiness probes will be defined correctly (this is what you failed to do when you manually touched the ConfigMap and why you should only configure Kiali via the Kiali CR).

**Reply by rohit131989** [reply]
Much appreciated !

**Comment by jmazzitelli** [comment]
Oh, and when you say `I did following changes in the config map of kiali`, do not do that :)

You also didn't say how you installed Kiali, but if you installed via the operator changing the Kiali CR is how you update the config, NOT the ConfigMap (the kiali.io docs explain this)

If you installed via helm, you set it through the values.

For one, if you change the ConfigMap you have to manually restart the pod yourself. Secondly, you may not be changing all the values that you need to change if you manually touch the ConfigMap. This is why we tell people never touch resources created by the operator - only edit the Kiali CR. (for helm, do a `helm update` with changes to the values yaml).
